### PR TITLE
Secure the takeover script

### PR DESCRIPTION
**Pull request by cevian**
Make sure that all objects made part of the extension are owned by
the object owner. Objects owned by someone else can easily be exploited
later. Do this automatically by traversing the depend tree.

### CODE REVIEW

File: sql/promscale--0.0.0.sql [hunk 1]
```diff
@@ -505,3 +506,361 @@ CREATE TYPE _prom_ext.GapfillDeltaTransition (
...
+                'Object ' || pg_describe_object(classid, objid, objsubid) || 'is owned by ' || owner::text
+        )
+        into errors
+        from with_owner
+        where (
+            --either the owner is the object owner
+            (owner = obj_owner_oid)
+            OR
+            --or prom_admin (these are for metric and series tables)
+            (owner = (SELECT oid FROM pg_roles WHERE rolname = 'prom_admin') AND classid IN ('pg_class'::regclass, 'pg_type'::regclass))
```
Comment (jgpruitt) [inline sql/promscale--0.0.0.sql:608]: why is pg_type considered here? metric and series tables would both be pg_class, right?

**Review by jgpruitt — APPROVED** [review]
just one question

File: sql/promscale--0.0.0.sql (hunk 1 above)
Comment (cevian) [reply sql/promscale--0.0.0.sql:608]: each pg_class creates a corresponding pg_type internally, always. It's for the record type.

**Review by JamesGuthrie — CHANGES_REQUESTED** [review]
In general it looks good to me. I tested it with a few simple scenarios to validate that it does what's expected:

- change owner of existing function to other role
- change owner of existing view to other role
- add function with other owner
- add operator with other owner
- add table with other owner

In case anybody else would like to try the test script, it's available in a gist here: https://gist.github.com/JamesGuthrie/c16a18defb28b40141ff904dce9a37b3

**Review by JamesGuthrie — APPROVED** [review]

**Comment by cevian** [comment]
passed in https://github.com/timescale/promscale/actions/runs/2229919658

**Pull request merged**